enhancements/authentication/external-oidc-additional-identity-information-sources.md

```diff
@@ -0,0 +1,804 @@
+    the ability for users to authenticate with their cluster.
+    This may mean some unexpected behaviors, but at least users can still
+    access the cluster.
+    Alternatively, there is always the break-glass scenario of continuing to
+    use x509 certificate based authentication in the event of an outage.
+
+4. Mixing authentication responsibilities between the KAS and new webhook authenticator
+    * **Mitigation**: The webhook authenticator is an implementation detail that
+    users do not need to concern themselves with beyond troubleshooting purposes.
+    To an end-user, they will just make configuration changes on a central API
+    and the platform operators will handle the correct configuration of platform components.
+
+5. Additional resource consumption
+    * **Mitigation**: None. This is a side effect of needing to generically support
+    fetching of user identity information from an external source. This is no worse
+    than the integrated OAuth server running. In the future, the intention is for
+    this to get contributed back upstream to be native in the Kubernetes API server
+    so an additional component is no longer necessary.
+
+6. Maintenance burden
+    * **Mitigation**: While it is a maintenance burden to re-implement and keep up-to-date
+    the logic that exists today in the Kubernetes API server, this is done with the intention
+    of contributing this back to the upstream to be handled natively by the KAS.
+    In the event this isn't accepted upstream, this approach has a better maintenance cost than
+    maintaining specialized logic for handling the nuances of every identity provider our customers
+    use. Kubernetes has a history of maintaining stability and backwards compatibility where
+    a per-identity-provider approach to solving this problem is likely to be more brittle.
+
+7. Credential security and rotation
+    * **Mitigation**: This is an implementation detail, but we can ensure that the
+    webhook authenticator can handle changes to the files that it references for
+    this information. Any secrets will be mounted as a volume to some file path.
+
+### Drawbacks
+
+The main drawbacks of this approach are:
+- Technical complexity and development time/cost
+- Yet another middleman component for authentication (we are trying to move away from the OAuth server, this moves the needle back towards that state)
+- Users have more complexity to deal with for sourcing external claim information as opposed to a tailored solution.
+- If contributed back upstream, we may be stuck supporting an API model that doesn't translate well.
+
+While not perfect, the proposed approach was chosen because it:
+- Is less maintenance burden to implement a generic approach than maintaining a specialized implementation per identity provider.
+- Has potential to be contributed back upstream.
+- Enables a use case that many of our enterprise customers need.
+
+## Alternatives (Not Implemented)
```

Comment:
We're also rejecting carrying a patch that adds a new authenticator in-process, instead of running a webhook service, right?